### PR TITLE
Threaded binary tree

### LINKED ISSUE #2009: Single Threaded binary tree
Implement single threaded binary tree (TBT) using cpp.

### PR DESCRIPTION
**Pull request by vedangi163**
Fixes #2009 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Add new file for Threaded binary tree. 

#### Languages Used:
- Cpp

#### Files Added:
- Threded_BT.cpp
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
+/*Implement single threaded binary tree and do inorder travrsal on it.
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Add a space after `/*`

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    Single Threaded Binary Tree : each node is threaded towards either the in-order 
+    predecessor or successor (left or right) means all right null pointers will 
+    point to inorder successor OR all left null pointers will point to inorder predecessor.
+NOTE:  
+    The right link of rightmost node from right subtree is always NULL.
+*/
+#include <iostream>
+using namespace std;
+class Node
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Add a new line after import statements.

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+public:
+    Node()
+    {	
+     left = right = NULL;
+     data = 0;
+     rtflag = 0;
+    }
+friend class Tree;
+};
+class Tree
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: There should be 2 new-lines between two classes.

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+{
+    public:
+        int flag,key,level,top,d;
+	char ch;
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Please follow correct indentation. Use 4 spaces for 1 indent.

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	Node *root,*q,*n;
+	Node* s[10];
+	Tree()
+	{
+            flag = key = level = d = 0;
+            top = -1;
+	    root = n = q = NULL;
+	    ch = 'n';
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Fix the indentation. In the whole file.

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+     rtflag = 0;
+    }
+friend class Tree;
+};
+class Tree
+{
+    public:
+        int flag,key,level,top,d;
+	char ch;
+	Node *root,*q,*n;
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Add space after `,`.

File: Threaded_Binary_Tree/Threaded_BT.cpp [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+	    root = n = q = NULL;
+	    ch = 'n';
+	}
+	
+	void addNode()
+	{
+	 do
+	 {
+	     flag = 0;
+             cout<<"\n Enter data : ";
```
Comment (SakshiShreya) [inline Threaded_Binary_Tree/Threaded_BT.cpp]: Add space around `>>` and `<<`.

**Comment by somya-kapoor** [comment]
Due to no activity for more than 7 days, I am closing this PR.

**Pull request closed without merging**